plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/ElasticsearchClient.java

```diff
@@ -520,38 +545,142 @@ public boolean indexExists(String index)
 
     public IndexMetadata getIndexMetadata(String index)
     {
+        try {
+            return indexMetaDataCache.getUnchecked(index);
+        }
+        catch (UncheckedExecutionException e) {
+            throwIfInstanceOf(e.getCause(), TrinoException.class);
+            throw e;
+        }
+    }
+
+    private IndexMetadata loadIndexMetadata(String index)
+    {
+        LOG.debug("IndexMetadata for index - '%s' not found in cache. Hence fetching from elasticsearch...", index);
+
         String path = format("/%s/_mappings", index);
 
         return doRequest(path, body -> {
             try {
-                JsonNode mappings = OBJECT_MAPPER.readTree(body)
-                        .elements().next()
-                        .get("mappings");
-
-                if (!mappings.elements().hasNext()) {
-                    return new IndexMetadata(new IndexMetadata.ObjectType(ImmutableList.of()));
+                Iterator<JsonNode> indicesIterator = OBJECT_MAPPER.readTree(body).elements();
+                if (maxNumberOfIndicesForAliasSchema > 0) {
+                    indicesIterator = Iterators.limit(indicesIterator, maxNumberOfIndicesForAliasSchema);
```

Comment:
This is actually configurable. If `maxNumberOfIndicesForAliasSchema` is 0 then it merges all of them. But if users want to configure to bypass the time to merge, this comes into play. 

Probably we can also have a configurable list of indices to be considered for merging. Thoughts?

Reply:
@martint @aalbu @bitsondatadev Thoughts?

Reply:
@martint we had a discussion a while back to make some limit on this provided it was configurable. I think we discussed making this percentage rather than a hard limit but a hard limit seems fine to me. It is much safer as there could be a massive amount of indexes that fall under an alias but I agree it is likely rarer to have this as it's not good practice so 0 is a good default. 

I definitely think this setting should exist. So I vote to keep it.

Reply:
Done.